tests/validation/cattlevalidationtest/core/test_github.py

```diff
@@ -18,92 +15,161 @@ def __init__(self, *args, **kwargs):
 
 
 if_github = pytest.mark.skipif(os.environ.get('API_AUTH_GITHUB'
-                               '_TEST_USER') is None,
-                               reason='API_AUTH_GITHUB_TEST_USER is not set')
+                               '_CLIENT_SECRET') is None,
+                               reason='API_AUTH_GITHUB'
+                                      '_CLIENT_SECRET is not set')
 
 BASE_URL = cattle_url() + '/v1/'
 URL = BASE_URL + 'schemas'
 
 
-@pytest.fixture(scope='module')
-def github_request_code(user=None, pw=None):
+@pytest.fixture(scope='session', autouse=True)
+def config():
+    needed_vars = [
+        'API_AUTH_GITHUB_TEST_USER',
+        'API_AUTH_GITHUB_TEST_PASS',
+        'API_AUTH_GITHUB_CLIENT_ID',
+        'API_AUTH_GITHUB_CLIENT_SECRET',
+        'API_AUTH_RANCHER_TEST_PASS',
+
+    ]
+    for a in needed_vars:
+        if os.getenv(a, None) is None:
+            raise Exception('Please set ' + a + ' in the environment')
+    config = {}
+    config['username'] = os.getenv('API_AUTH_GITHUB_TEST_USER', None)
+    config['password'] = os.getenv('API_AUTH_GITHUB_TEST_PASS', None)
+    config['phantomjs_port'] = int(os.getenv('PHANTOMJS_WEBDRIVER_PORT', 4444))
+    config['phantomjs_bin'] = os.getenv('PHANTOMJS_BIN',
+                                        '/usr/local/bin/phantomjs')
+    assert config['phantomjs_bin'] is not None
+    config['client_id'] = os.getenv('API_AUTH_GITHUB_CLIENT_ID', None)
+    config['client_secret'] = os.getenv('API_AUTH_GITHUB_CLIENT_SECRET', None)
+    config['users'] = {}
+    config['users']['1'] = {
+        'password': os.getenv('API_AUTH_RANCHER_TEST_PASS', None),
+        'username': os.getenv('API_AUTH_RANCHER_TEST_USER_1', 'ranchertest01')
+    }
+    config['users']['2'] = {
+        'password': os.getenv('API_AUTH_RANCHER_TEST_PASS_2', None),
+        'username': os.getenv('API_AUTH_RANCHER_TEST_USER_2', 'ranchertest02')
+    }
+    return config
 
-    username = os.getenv('API_AUTH_GITHUB_TEST_USER', None)
-    password = os.getenv('API_AUTH_GITHUB_TEST_PASS', None)
-    phantomjs_port = int(os.getenv('PHANTOMJS_WEBDRIVER_PORT', 4444))
-    phantomjs_bin = os.getenv('PHANTOMJS_BIN', '/usr/local/bin/phantomjs')
 
+@pytest.fixture(scope='module')
+def github_request_code(config, cattle_url, admin_client, request, user=None):
+    def fin():
+            admin_client.create_githubconfig(enabled=False)
+    request.addfinalizer(fin)
+    username = config['username']
+    password = config['password']
     if user is not None:
-        username = user
+        username = user['username']
+        password = user['password']
 
-    if pw is not None:
-        password = pw
+    assert config is not None
```

Comment:
Is this necessary?  or maybe should be moved further up since we already try to get 'username' and 'password' above